### ISSUE TITLE
Stabilize Qwen3.5 397B repeated evals by deterministic MoE routing

### ISSUE THREAD

**Issue by raikonenfnu** [raikonenfnu/tokenspeed#20]
## Summary

While chasing the remaining TP=4 Qwen3.5 397B same-server repeat drift, the failure stopped looking like a GDN/page-table/cache problem and narrowed to MoE routing determinism.

The important observation was that the inputs to routing could be identical while the selected expert ids were not. In the layer-0 MoE probe, `hidden_states`, `router_logits`, and `topk_weights` matched across repeated requests, but `topk_ids` differed. That is enough to change the expert output, then the next layer sees different hidden states and the request diverges over a long decode.

## Concrete example

The earlier probes showed:

```text
hidden_local_in exact
router_local exact
hidden_after_pre_mlp_comm exact
router_after_pre_mlp_comm exact
shared_output exact
topk_router_logits exact
topk_weights exact
topk_ids differ substantially
experts_out_before_shared differs afterward
```

That is the signature of a tie/order issue rather than a math-input issue. Qwen3.5 on AMD selects the Triton MXFP4 precomputed-topk MoE path, so Python/native top-k computes the expert ids first and Triton consumes those ids directly. If equal router scores resolve to different ids between runs, the MoE kernel is doing what it was asked to do, but the request is no longer deterministic.

A small version of the same problem is a row of equal scores. With an unstable top-k contract, `[0, 1]`, `[1, 0]`, or another valid top-k ordering can all be defensible for equal values. For inference, that ambiguity is not harmless: expert id order is part of the computation.

## Implemented fix

The native MoE top-k path now resolves ties with stable descending ordering:

```text
scores -> stable descending sort -> first k experts
```

This means exact ties choose the lower/earlier expert id consistently. The same helper is used for both the normal Qwen-style path and the correction-bias path, where the biased score chooses the ids but the original softmax score still supplies the weights.

The intuition is simple: if router scores are identical, the selected expert set should be a deterministic function of `(score, expert_id)`, not a function of backend tie behavior. That makes the precomputed-topk contract stable before the Triton MoE kernel runs.

## Eval isolation fix

The AIME reproducer also exposed a separate source of perturbation. During long streamed eval batches, SMG health checks were entering the engine as real `HEALTH_CHECK_*` generation requests, temporarily changing the batch from 8 running requests to 9. That can alter scheduling/capture/batching behavior during a deterministic eval.

`ts serve` now disables SMG health checks by default, matching the PD launcher's explicit `--disable-health-check` behavior. Liveness should not inject model work into an active benchmark batch.

## CI/eval config fix

Qwen3.5 397B MXFP4 AIME through the OpenAI API also needs thinking disabled via:

```json
{"extra_body":{"chat_template_kwargs":{"enable_thinking":false}}}
```

Without this, the local AIME limit=8 run scored `0.75` and generated much longer outputs. With thinking disabled, the same-server runs were back at/above the intended threshold.

## Local validation

Short repeated prompt, same TP=4 397B server:

```text
5/5 identical hashes
```

Long 2048-token repeated prompt, same TP=4 397B server:

```text
3/3 identical hashes
```

AIME25 limit=8, eval-batch-size=8, stream, max_tokens=16384, same TP=4 397B server, thinking disabled:

```text
Run 1: score 0.875, wall 484.19s, TTFT 625.17ms, TPOT 22.50ms, throughput 38.58 tok/s
Run 2: score 1.000, wall 495.02s, TTFT 375.85ms, TPOT 23.36ms, throughput 35.67 tok/s
Run 3: score 1.000, wall 477.72s, TTFT 343.42ms, TPOT 22.26ms, throughput 39.20 tok/s
```

Focused tests:

```text
pytest -q test/runtime/test_moe_topk.py test/runtime/test_update_block_table_capacity_clamp.py
7 passed

pytest -q test/cli/test_serve_smg_unit.py
37 passed

pre-commit run --all-files
passed
```

## Remaining note

The full AIME long-form text is still not bit-identical across repeated runs, even when the extracted answers are correct. The practical regression that motivated this issue was the progressive same-server accuracy degradation; the deterministic top-k change plus eval isolation removes the concrete sources found so far and keeps the limit=8 AIME score at or above the `0.875` threshold in local runs.
